File: bharat-railway-backend/src/main/java/com/bharatrailway/booking/application/dto/BookingRequest.java
/*
 * Project: Bharat Railway Booking System
 * Module: bharat-railway-backend
 * Branch: feature/backend-developer-hitanshu
 * Developer: Chandra Shekhar Bansal
 * Assisted by: DeepSeek (AI Scribe)
 * Date: 2026-09-01
 * Version: 0.1.0-SNAPSHOT
 *
 * Description:
 * Booking request DTO for creating a new booking with passengers.
 */

package com.bharatrailway.booking.application.dto;

import java.time.LocalDate;
import java.util.List;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public class BookingRequest {

    @NotNull(message = "User ID is required")
    private Integer userId;

    @NotBlank(message = "Train number is required")
    @Size(max = 5, message = "Train number must not exceed 5 characters")
    private String trainNumber;

    @NotBlank(message = "Source station code is required")
    @Size(max = 5, message = "Source station code must not exceed 5 characters")
    private String sourceStationCode;

    @NotBlank(message = "Destination station code is required")
    @Size(max = 5, message = "Destination station code must not exceed 5 characters")
    private String destinationStationCode;

    @NotNull(message = "Journey date is required")
    private LocalDate journeyDate;

    @NotEmpty(message = "At least one passenger is required")
    @Valid
    private List<PassengerRequest> passengers;

    public BookingRequest() {
    }

    public Integer getUserId() {
        return userId;
    }

    public void setUserId(Integer userId) {
        this.userId = userId;
    }

    public String getTrainNumber() {
        return trainNumber;
    }

    public void setTrainNumber(String trainNumber) {
        this.trainNumber = trainNumber;
    }

    public String getSourceStationCode() {
        return sourceStationCode;
    }

    public void setSourceStationCode(String sourceStationCode) {
        this.sourceStationCode = sourceStationCode;
    }

    public String getDestinationStationCode() {
        return destinationStationCode;
    }

    public void setDestinationStationCode(String destinationStationCode) {
        this.destinationStationCode = destinationStationCode;
    }

    public LocalDate getJourneyDate() {
        return journeyDate;
    }

    public void setJourneyDate(LocalDate journeyDate) {
        this.journeyDate = journeyDate;
    }

    public List<PassengerRequest> getPassengers() {
        return passengers;
    }

    public void setPassengers(List<PassengerRequest> passengers) {
        this.passengers = passengers;
    }
}

File: bharat-railway-frontend/lib/models/booking_request.dart
/*
 * Project: Bharat Railway Booking System
 * Module: bharat-railway-frontend
 * Branch: feature/frontend-developer-chandrashekhar
 * Developer: Chandra Shekhar Bansal
 * Date: 2026-09-01
 * Version: 1.0.0
 *
 * Description:
 * Booking request model for creating a new booking.
 */

import 'passenger_request.dart';

class BookingRequest {
  final int userId;
  final String trainNumber;
  final String sourceStationCode;
  final String destinationStationCode;
  final String journeyDate;
  final List<PassengerRequest> passengers;

  BookingRequest({
    required this.userId,
    required this.trainNumber,
    required this.sourceStationCode,
    required this.destinationStationCode,
    required this.journeyDate,
    required this.passengers,
  });

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'trainNumber': trainNumber,
      'sourceStationCode': sourceStationCode,
      'destinationStationCode': destinationStationCode,
      'journeyDate': journeyDate,
      'passengers': passengers.map((p) => p.toJson()).toList(),
    };
  }
}